src/angular-intro.js

```diff
@@ -41,33 +41,37 @@ ngIntroDirective.directive('ngIntroOptions', ['$timeout', function ($timeout) {
                 
                 if (scope.ngIntroOncomplete) {
                     intro.oncomplete(function() {
-                        $timeout(scope.ngIntroOncomplete.bind(this, scope));
+                        scope.ngIntroOncomplete.call(this, scope);
+                        scope.$digest();
                         navigationWatch();
                     });
                 }
 
                 if (scope.ngIntroOnexit) {
                     intro.onexit(function() {
-                        $timeout(scope.ngIntroOnexit.bind(this, scope));
-                        navigationWatch();
```

Comment:
Does the `navigationWatch()` need to be removed here, was it interfering with something?